src/sql/migrations.rs

```diff
@@ -596,6 +596,13 @@ CREATE INDEX smtp_messageid ON imap(rfc724_mid);
         )
         .await?;
     }
+    if dbversion < 92 {
+        sql.execute_migration(
+            "CREATE TABLE sending_domains(domain TEXT PRIMARY KEY, dkim_works INTEGER DEFAULT 0);",
```

Comment:
> Maybe we also want to require valid dkim check for a month or so, otherwise if server setup previously working breaks at some point, Delta Chat will never again accept autocrypt changes.

To do this, we would need to store the _first_ time we received a valid dkim check though.

And if we want to allow key changes again after dkim was invalid for some time, we need to store the first time we received an _invalid_ dkim check after dkim was valid, and e.g. 0=dkim was valid in the last email, 1=dkim never was valid

Reply:
Ah OK, so that the trade-off is, we just unconditionally accept key changes after we didn't get a message from a domain for a month.

While I don't have an opinion on this, in one of our discussions @hpk42 said that he would like to avoid "magical" time spans - but we could also start storing this timestamp now in order to make the migration easier if it turns out that it's needed while testing this.

Reply:
> But if we don't add some way to recover here

I always assumed/hoped that the way to recover is scanning the other person's QR code, but it turns out, I was wrong, it doesn't. And from looking at the securejoin code (which I don't know and never touched myself though), it looks like it's not easy to fix.

Anyway, let's keep to `dkim_works` in this thread here - I added it in c5e93bf0 and 8aaea481. For setting dkim_works, we probably have to use the message's time (because otherwise receiving an old message would also bump dkim_works), while for getting dkim_works, we have to use the system time (because using the message time would mean that an attacker can circumvent authres-checking by setting a Date: in the future, so that we think that it's a long time ago that DKIM worked.